pkg/kv/kvserver/loqrecovery/utils.go

```diff
@@ -152,6 +265,38 @@ func (i rangeReplicaChange) Span() roachpb.Span {
 	return i.span
 }
 
+type allReplicasLost struct {
+	rangeID roachpb.RangeID
+	span    roachpb.Span
+}
+
+func (i allReplicasLost) String() string {
+	return fmt.Sprintf("range does not have any replicas remaining\n  r%d %v",
+		i.rangeID,
+		i.span)
+}
+
+func (i allReplicasLost) Span() roachpb.Span {
+	return i.span
+}
+
+type rangeMetaMismatch struct {
```

Comment:
Would be good with a comment on this and the above clarifying what exactly they represent.